core/task_mapper/heft/heft.py

```diff
@@ -345,7 +353,9 @@ def run(self, task_mapper):
                 self.processors[p].time_line.append(Duration(task.number, task.ast, task.aft))
                 self.processors[p].time_line.sort(key=functools.cmp_to_key(lambda x, y: cmp(x.start, y.start)))
 
-        self.duplicate()
+        if task_mapper == "heft_duplicate" or task_mapper == "heft_dup_no_comm_cost":
+            self.duplicate()
+
         self.reschedule()
```

Comment:
reschedule() is a part of HEFT duplication (duplicate() is calculating schedule, reschedule() is actually doing it), thus should be put under 'if'

Reply:
fixed